src/libsync/vfs/cfapi/vfs_cfapi.cpp

```diff
@@ -370,30 +381,30 @@ void VfsCfApi::onHydrationJobFinished(HydrationJob *job)
 {
     Q_ASSERT(d->hydrationJobs.contains(job));
     qCInfo(lcCfApi) << "Hydration job finished" << job->requestId() << job->folderPath() << job->status();
-    emit hydrationRequestFinished(job->requestId(), job->status());
-    d->hydrationJobs.removeAll(job);
-    if (d->hydrationJobs.isEmpty()) {
-        emit doneHydrating();
-    }
+    emit hydrationRequestFinished(job->requestId());
 }
 
-void VfsCfApi::onHydrationJobCanceled(HydrationJob *job)
+int VfsCfApi::finalizeHydrationJob(const QString &requestId)
```

Comment:
we can still have a race condition when calling this method because if I understand correctly when we are in `cfApiFetchDataCallback` we get a different thread for each call to the callback

Reply:
yes you are right I adjusted the code